baseapp/baseapp.go

```diff
@@ -337,7 +337,7 @@ func handleQueryApp(app *BaseApp, path []string, req abci.RequestQuery) (res abc
 		}
 
 		// Encode with json
-		value := codec.Cdc.MustMarshalBinary(result)
+		value := codec.Cdc.MustMarshalBinaryLengthPrefixed(result)
```

Comment:
Wasn't the conclusion that most things (except maybe IAVL key/values) need not be length prefixed? I did I misread?

/cc @ValarDragon

Reply:
yeah I don't think this should be length prefixed. This could be addressed separately though.